core/ecschema-editing/src/Editing/Constants.ts

```diff
@@ -43,27 +52,34 @@ export class Constants {
     if (displayLabel)
       newConstant.setDisplayLabel(displayLabel);
 
-    return { itemKey: newConstant.key };
+    return newConstant.key;
   }
 
   /**
    * Creates a Constant through a ConstantProps.
    * @param schemaKey a SchemaKey of the Schema that will house the new object.
    * @param relationshipProps a json object that will be used to populate the new RelationshipClass. Needs a name value passed in.
    */
-  public async createFromProps(schemaKey: SchemaKey, constantProps: ConstantProps): Promise<SchemaItemEditResults> {
+  public async createFromProps(schemaKey: SchemaKey, constantProps: ConstantProps): Promise<SchemaItemKey> {
     const schema = await this._schemaEditor.getSchema(schemaKey);
     if (schema === undefined)
-      return { errorMessage: `Schema Key ${schemaKey.toString(true)} not found in context` };
+      throw new ECEditingError(ECEditingStatus.SchemaNotFound, `Schema Key ${schemaKey.toString(true)} not found in context`);
 
     if (constantProps.name === undefined)
-      return { errorMessage: `No name was supplied within props.` };
+      throw new ECEditingError(ECEditingStatus.SchemaItemNameNotSpecified, `No name was supplied within props.`);
 
-    const newConstant = (await schema.createConstant(constantProps.name));
-    if (newConstant === undefined)
-      return { errorMessage: `Failed to create class ${constantProps.name} in schema ${schemaKey.toString(true)}.` };
+    let newConstant: MutableConstant;
+    try {
+      newConstant = (await schema.createConstant(constantProps.name)) as MutableConstant;
+    } catch (e) {
+      if (e instanceof ECObjectsError && e.errorNumber === ECObjectsStatus.DuplicateItem) {
+        throw new ECEditingError(ECEditingStatus.SchemaItemNameAlreadyExists, `Constant ${constantProps.name} already exists in the schema ${schema.fullName}.`);
+      } else {
+        throw new ECEditingError(ECEditingStatus.Unknown, `Failed to create Constant ${constantProps.name} in schema ${schema.fullName}.`);
+      }
+    }
```

Comment:
Since we have that block repeating very often, I wonder if we can capsule it away. I could imagine something like this:
```
const newConstant = await schema.createConstant(constantProps.name)
    .catch(handleItemCreationError("Constant", constantProps.name), schema.fullName));
```

the handleItemCreationError has the three variable parts of the repeating code block and would then look like this:
```
function handleError(type: string, name: string, schema: string) {
  return (e: unknown) => {
     if (e instanceof ECObjectsError && e.errorNumber === ECObjectsStatus.DuplicateItem) {
        throw new ECEditingError(ECEditingStatus.SchemaItemNameAlreadyExists, `${type} ${name} already exists in the schema ${schema}.`);
      } else {
        throw new ECEditingError(ECEditingStatus.Unknown, `Failed to create ${type} ${name} in schema ${schema}.`);
      }
  }
}
```

So if createConstant fails, the promise calls the catch callback, there we can check the type and rethrow an updated error up again to the promise which will eventually fail and return the error back to the caller.

Reply:
My refactor should address all this redundancy.